paimon-common/src/main/java/org/apache/paimon/format/FormatReaderFactory.java

```diff
@@ -33,4 +36,19 @@ public interface FormatReaderFactory extends Serializable {
 
     RecordReader<InternalRow> createReader(FileIO fileIO, Path file, int poolSize)
             throws IOException;
+
+    RecordReader<InternalRow> createReader(
```

Comment:
Do we need to introduce these methods?

I think we can just handle return value from `RecordReader`.
```
if (it is a partitioned table) {
  if (it is a `ColumnarRowIterator`) {
    then use `VectorMappingUtils.wrap`;
  } else {
    use a PartitionedRow to wrap it;
  }
}
```